host/app/components/card-editor.gts

```diff
@@ -239,10 +242,23 @@ export default class Preview extends Component<Signature> {
       throw new Error(`could not save file, status: ${response.status} - ${response.statusText}. ${await response.text()}`);
     }
     let json = await response.json();
-    if (json.data.links?.self && this.args.onSave) {
+    
+    // reset our dirty checking to be detect dirtiness from the
+    // current JSON to reflect save that just happened
+    this.initialCardData = this.currentJSON;
+
+    if (json.data.links?.self) {
       // this is to notify the application route to load a
       // new source path, so we use the actual .json extension
-      this.args.onSave(json.data.links.self + '.json');
+      this.doSave(json.data.links.self + '.json');
+    }
+  }
+
+  doSave(path: string) {
+    if (this.args.onSave) {
+      this.args.onSave(path);
+    } else {
+      this.setFormat('isolated')
```

Comment:
This was just a little tweak to make things act more reasonably. so that dirtiness was reset after you save a card, and that we transition back to the view format after save (otherwise it looks like nothing happened).